### ISSUE TITLE
Blood Bowl 2 (236690)

### ISSUE THREAD

**Issue by Gepp35** [ValveSoftware/Proton#1978]
# Compatibility Report
- Name of the game with compatibility issues: Blood Bowl 2
- Steam AppID of the game: 236690

## System Information
- GPU: AMD RAVEN (ryzen 7 pro 2700U/ radeon vega)
- Driver/LLVM version: mesa 19.0.0-devel (2018-11-15 bionic oibaf), DRM 3.27.0, LLVM 7.0.1
- Kernel version: kernel 4.19.0-041900-generic (ubuntu LT 18.04 uptodate)
- Link to full system information report as [Gist](https://gist.github.com/): OpenGL 4.5
- Proton version: proton 3.16-4 beta

## I confirm:
- [x ] that I haven't found an existing compatibility report for this game.
- [x ] that I have checked whether there are updates for my system available.

## Symptoms 
Game doesn't launch with steam proton. It launches using Lutris until before the game menu. Seems to work with Nvidia video card.

**Comment by kisak-valve** [comment]
Hello @Gepp35, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and drag and drop the generated $HOME/steam-$APPID.log into the comment box.

**Comment by Gepp35** [comment]
[steam-236690.log](https://gist.github.com/kisak-valve/6955b73d104ab4975fa19fb87439ed8d)

**Comment by kisak-valve** [comment]
Bloob Bowl 2 writes gigabytes of log spam to ~/.xsession-errors

Issue transferred from https://github.com/ValveSoftware/Proton/issues/2457.
@jwakely posted on 2019-03-25T20:30:45:

# Compatibility Report
- Name of the game with compatibility issues: Blood Bowl 2
- Steam AppID of the game: 236690

## System Information
- GPU: GTX 1060 6gb
- Driver/LLVM version:  nvidia 418.43
- Kernel version:  4.20.16
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/jwakely/875fc2b863129f1e1118fc7a7bb9490a
- Proton version: 3.16-8 Beta

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.



After starting the game and then quitting as soon as the main menu appeared, the `steam-236690.log` file was 246MB and so github won't allow me to upload it!

This is a log file created by interrupting the game as soon as the first "Viewport hasn't been setted" errors were written to the log:

[steam-236690.log](https://github.com/ValveSoftware/Proton/files/3005350/steam-236690.log)


## Symptoms

The game writes the line "`Viewport hasn't been setted`" to the file `~/.xsession-errors` thousands of times per minute. Over the space of a few days it logged that line 3.8 billion times, and filled my home partition (which previously had 100gb free).

## Reproduction

In a terminal run `tail -f ~/.xsession-errors | grep setted` and then start the game. Watch the logs fly by ...

**Comment by jwakely** [comment]
I forgot to mention in my report that the game actually runs **really well** for me, and I didn't even realise there was anything wrong until my home partition filled up for the second time in a week. After that I went looking for the problem and found that Blood Bowl 2 was doing all this logging. The [steam-236690.log](https://github.com/ValveSoftware/Proton/files/3005350/steam-236690.log) file shows loads of exceptions being raised, but this doesn't affect game play at all.

**Comment by jwakely** [comment]
~~As a workaround, I have to ` rm ~/.xsession-errors` and reboot before the file fills my disk again.~~
The simplest workaround is to Exit t he Steam client and restart it from a terminal window, redirecting the output to `/dev/null`:

    steam >/dev/null

That way nothing gets written to `.xsession-errors`

**Comment by aeikum** [comment]
Interesting. The game continually dumps that via OutputDebugString, so that's why your Steam log grows so fast. I have no idea how that message makes it into .xsession-errors, though... that doesn't happen on my Arch Linux machine.

Edit: Ah, it gets dumped to stdout. Probably whatever launches Steam is logging its output to .xsession-errors.

**Comment by jwakely** [comment]
It's the Xorg server that writes it there, because that's what the file is for: capturing the output of X Windows processes.

If I launch `steam` from an xterm then the output goes to that terminal, but that's not how I launch the steam client (and neither do most people, I assume).

**Comment by aeikum** [comment]
Interesting. We could redirect stdout to /dev/null by default. It's been brought up before (Steam also logs games' stdout to /tmp/dumps/, which will grow just like your xsession file).

**Comment by jwakely** [comment]
Could that redirecting be done on a game-by-game basis, e.g. via the launch options, so that output is only discarded for games known to spew too much?

**Comment by kisak-valve** [comment]
Hello @jwakely, it may be worth trying `> /dev/null` (or `> /dev/null 2>&1` to include stderr) in the game's launch options. This always has to be the last thing in the launch options.

**Comment by jwakely** [comment]
@kisak-valve yeah that works, thanks. That's a perfectly good solution as far as I'm concerned, I can play BB2 without needing to run `steam` from a terminal and without worrying about disks filling up. Everything else works well.

**Comment by jwakely** [comment]
Oh, except after switching away from the game with Alt-Tab and then back again, you need to press Alt once to re-enable the keyboard. I've seen a few comments saying that for different games on protondb.com. It's not a big deal, you just have to remember to press Alt again.
